tests/staticfiles_tests/test_storage.py

```diff
@@ -445,6 +445,14 @@ def test_missing_entry(self):
         # File exists on disk
         self.hashed_file_path(missing_file_name)
 
+    def test_doesnt_create_intermediate_files(self):
+
+        intermediate_file = 'cached/relative.ccc37118948a.css'
```

Comment:
Just to make sure, this test fails before the change? (It's hard to know with a "negative" test and a hard-coded hash).

Reply:
Yep, it does. When I copy/paste the test from this PR on master I get the following output:
```
$ ./runtests.py staticfiles_tests.test_storage.TestCollectionManifestStorage.test_doesnt_keep_intermediate_files
Testing against Django installed in '/path/to/django/django' with up to 8 processes
System check identified no issues (0 silenced).
F
======================================================================
FAIL: test_doesnt_keep_intermediate_files (staticfiles_tests.test_storage.TestCollectionManifestStorage)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/path/to/django/tests/staticfiles_tests/test_storage.py", line 453, in test_doesnt_keep_intermediate_files
    self.assertFalse(os.path.exists(collected_path))
AssertionError: True is not false

----------------------------------------------------------------------
Ran 1 test in 0.073s

FAILED (failures=1)
```

I can split this into two commits if you'd like to see the test fail here.